### ISSUE TITLE
 Move/copy a PDF file for an entry WITHOUT renaming (drag and drop)

### ISSUE THREAD

**Issue by Siedlerchr** [JabRef/jabref#7313]
Originally posted at https://discourse.jabref.org/t/move-copy-a-pdf-file-for-an-entry-without-renaming/2495
 
> When I used JabRef version 4.X, I could drag-and-drop (D&D) a PDF file to the entry of interest to move the PDF file to my JabRef file folder and to link it with the entry simultaneously. Version 5 lost this capability. My suggestion is to assign a modifier key to work with D&D to achieve Move/Copy without renaming (as follows for an example).
> Copy and rename : Ctrl + D&D
> Copy WITHOUT rename : Shift + Ctrl + D&D
> Move and rename : No key + D&D
> Move WITHOUT rename: Shift + D&D
> Link : ALT + D&D
> Please read below for more detail.
> 
> Entry Editor - JabRef explains the how combinations of D&D with modifier keys work as follows (for Windows, in my case);
> 
>     Move: Ctrl + D&D
>     Copy: Shift + Drag or no key + D&D
>     Link: Alt + D&D
> 
> With version 5.3 on Windows 10, what happens are different form aforementioned info, and actually go as follows;
> Copy and rename: Ctrl + D&D
> Move and rename: Shift + D&D, No key + D&D
> Link: ALT + D&D
> (I don’t find any combination to achieve “Move without rename”.)
> 
> Ctrl and Shift modifier’s functions are interchanged in the linked site. More importantly, both Move and Copy functions rename the file. I saw the functions of “Shift + D&D” and “No key + D&D” are redundant. I would suggest that changing “Shift + D&D” for “Move without rename” and “Ctrl + Shift + D&D” for “Copy without rename”.
> 
> I would appreciate if you consider this modification for new JabRef version.
> 
> Thank you.
> Masa

**Comment by Siedlerchr** [comment]
As the modifiers are OS dependent, we can maybe change copy to just copy without rename

**Comment by ilippert** [comment]
How about this:
in the preferences under "linked files", a dropdown menu offers diverse default actions for dragging file onto entry.
Then, for instance, I would define as a default action "copy without rename"; somebody else "move with rename".
I would presume that individual users have specific file handling routines that are highly tight to their work practices and requirements. So, allow the user to select (and change) the default.
Of course, a fresh JabRef configuration would still have to define a default.
From my perspective I see these options:

- always ask me
- Copy and rename 
- Copy WITHOUT rename
- Move and rename 
- Move WITHOUT rename
- Link WITHOUT rename
- Link and rename

**Comment by masa312f** [comment]
I strongly agree to allow users to choose/assign the default drag-&-drop action and functions of modifier keys in the Preferences.

**Comment by ilippert** [comment]
maybe label this low priority? but still good idea.

**Comment by masa312f** [comment]
Hi illippert,
Thank you for your support. I still want to have this option for linking/transferring files to the entry of interest.
I hope if I could have a knowledge to modify the software.....

**Comment by ThiloteE** [comment]
This issue here is still relevant, as https://discourse.jabref.org/t/drag-and-drop-of-pdfs/3362 shows.

**Comment by ThiloteE** [comment]
Related: https://github.com/JabRef/jabref/issues/6742
